src/coreclr/jit/gentree.cpp

```diff
@@ -22087,11 +22087,11 @@ GenTree* Compiler::gtNewSimdNarrowNode(var_types   type,
 
                 GenTree* vecCon2 = gtCloneExpr(vecCon1);
 
-                tmp1 = gtNewSimdHWIntrinsicNode(type, op1, vecCon1, NI_SSE2_And, simdBaseJitType, simdSize,
+                tmp1 = gtNewSimdHWIntrinsicNode(type, op1, vecCon1, NI_AVX2_And, simdBaseJitType, simdSize,
```

Comment:
This wasn't "failing" because the `gtType` and `simdSize` were right and the XMM vs YMM instruction remains the same so everything "worked" but wasn't the right intrinsicId.